### PR TITLE
Update implicit availability to match spec changes

### LINKED ISSUE #10103: Handle implicit availability spec changes
In the latest [implicit tiling schema draft](https://github.com/CesiumGS/3d-tiles/tree/extension-revisions-implicit/extensions/3DTILES_implicit_tiling/schema/subtree), the way multiple contents is handled is a little different. instead of storing things in the subtree's `3DTILES_multiple_contents` extension, it'll now be part of the main subtree JSON. We do need to support both for backwards compatibility.

Old subtree JSON looks like this: [see old spec](https://github.com/CesiumGS/3d-tiles/tree/main/extensions/3DTILES_multiple_contents#implicit-tiling)

```jsonc
// Subtree JSON
{
  "tileAvailability": {
    "bufferView": 0
  },
  "extensions": {
    "3DTILES_multiple_contents": {
      "contentAvailability": [
        {
          "bufferView": 1
        },
        {
          "bufferView": 2
        }
      ]
    }
  }
}
```

New subtree JSON will look like this: (note that `bufferView` is now called `bitstream`. See also the [latest schema](https://github.com/CesiumGS/3d-tiles/tree/extension-revisions-implicit/extensions/3DTILES_implicit_tiling/schema/subtree)

```jsonc
// Subtree JSON
{
  "tileAvailability": {
    "bitstream": 0
  },
  "contentAvailability": [
    {
      "bitstream": 1
    },
    {
      "bitstream": 2
    }
  ]
}
```

- [x] in `ImplicitSubtree.js`, update the parsing for `contentAvailability` to find the availability JSON in 3 possible ways:
  - old schema may have `subtreeJson.contentAvailability` as a single object (no multiple contents)
  - old schema may have a `3DTILES_multiple_contents` extension as described in the example above
  - new schema will always have `subtreeJson.contentAvailability` be an _array_
- [x] In the same file, also handle the fact that `bufferView` was renamed to `bitstream` for tile, content, and subtree availabilities. (but only for availability, metadata will be handled in a different issue). Again, we need backwards compatibility.

CC @j9liu

### PR DESCRIPTION
**Pull request by j9liu**
Closes #10103.

This PR includes the following changes:
- Renamed the availability `bufferView` field to `bitstream` according to the specs, while allowing for backwards compatibility
- Added options to `ImplicitTilingTester` to produce different test cases, e.g. JSONs and fields for backwards compatibility
- Rewrote the unit tests for testing multiple content in `ImplicitSubtreeSpec`, since they were trivially passing
- Fixed a bug that wasn't properly retrieving multiple content in `ImplicitSubtree.prototype.contentIsAvailableAtCoordinates`
- Added ability to parse multiple contents in subtree files without requiring the extension itself
- Updated unit tests with a new test case

@ptrgags let me know how I should document this in `CHANGES.md`!

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Thanks for the pull request @j9liu!
* :heavy_check_mark: Signed CLA found.

Reviewers, don't forget to make sure that:

- [ ] Cesium Viewer works.
- [ ] Works in 2D/CV.

File: Specs/Data/Cesium3DTiles/Implicit/ImplicitMultipleContentsWithoutExtension/subtrees/0.0.0.json [hunk 1]
```diff
@@ -0,0 +1,29 @@
+{
+  "buffers": [
+    {
+      "byteLength": 8,
+      "uri": "0.bin"
```
Comment (ptrgags) [inline Specs/Data/Cesium3DTiles/Implicit/ImplicitMultipleContentsWithoutExtension/subtrees/0.0.0.json:5]: maybe rename the file `availability.bin` to be clearer about what it contains.

File: Source/Scene/ImplicitSubtree.js [hunk 2]
```diff
@@ -331,6 +331,8 @@ function initialize(subtree, json, subtreeView, implicitTileset) {
   if (hasExtension(subtreeJson, "3DTILES_multiple_contents")) {
     subtreeJson.contentAvailabilityHeaders =
       subtreeJson.extensions["3DTILES_multiple_contents"].contentAvailability;
+  } else if (Array.isArray(subtreeJson.contentAvailability)) {
```
Comment (ptrgags) [inline Source/Scene/ImplicitSubtree.js:334]: I would add comments explaining which parts are the current schema and which ones are from the older schema.

File: Source/Scene/ImplicitSubtree.js [hunk 3]
```diff
@@ -509,24 +511,41 @@ function preprocessBufferViews(bufferViewHeaders, bufferHeaders) {
 function markActiveBufferViews(subtreeJson, bufferViewHeaders) {
   let header;
   const tileAvailabilityHeader = subtreeJson.tileAvailability;
-  if (defined(tileAvailabilityHeader.bufferView)) {
+  if (defined(tileAvailabilityHeader.bitstream)) {
+    header = bufferViewHeaders[tileAvailabilityHeader.bitstream];
+  } else if (defined(tileAvailabilityHeader.bufferView)) {
```
Comment (ptrgags) [inline Source/Scene/ImplicitSubtree.js:516]: Same here, in this function comment explaining that `bufferView` was from an older schema draft.

**Review by ptrgags — COMMENTED** [review]
@j9liu just a few minor comments

As for the changelog, mention that implicit tiling availability is updated to the latest specification changes. 

Pretty soon we should organize these entries into a list about the 3D Tiles 1.1 spec, but let's wait until we have the full schema ready so we can link to it. We can do that in a future PR.

**Review by ptrgags — COMMENTED** [review]
@j9liu looks good, I'll commit the one rewording and I'll merge once CI passes.

**Pull request merged**